integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/VariableModesTest.kt

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.testapp.validation.examples
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.offset
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.designcompose.CustomizationContext
+import com.android.designcompose.DesignComposeCallbacks
+import com.android.designcompose.DesignDoc
+import com.android.designcompose.DesignSwitcherPolicy
+import com.android.designcompose.DesignVariableCollectionProvider
+import com.android.designcompose.LocalCustomizationContext
+import com.android.designcompose.OpenLinkCallback
+import com.android.designcompose.ParentComponentInfo
+import com.android.designcompose.TapCallback
+import com.android.designcompose.annotation.DesignComponent
+import com.android.designcompose.annotation.DesignDoc
+import com.android.designcompose.annotation.DesignVariableMode
+import com.android.designcompose.common.DocumentServerParams
+import com.android.designcompose.mergeFrom
+import com.android.designcompose.sDocClass
+import com.android.designcompose.serdegen.NodeQuery
+import com.android.designcompose.setCustomComposable
+import com.android.designcompose.setKey
+import com.android.designcompose.setOpenLinkCallback
+import com.android.designcompose.setTapCallback
+import com.android.designcompose.setVariableMode
+
+enum class LightDarkMode {
+    Light,
+    Dark,
+}
+
+enum class Theme(val themeName: String) {
+    Material("material-theme"),
+    MyTheme("my-theme"),
+}
+
+@DesignDoc(id = "HhGxvL4aHhP8ALsLNz56TP")
+interface VariablesTest {
+    @DesignComponent(node = "#stage")
+    fun Main(
+        // @DesignVariableCollection(node = "#stage") theme: Theme?,
+        @DesignVariableMode(node = "#Left", collection = "material-theme")
```

Comment:
This method of setting a mode uses an annotation to set the mode on a node. Note that you must also specify the collection (i.e. "theme", but called a colleciton in Figma), since each collection has a set of modes that can have the same name. We could alternatively remove this annotation in favor of allowing the developer to wrap their code around a compositionLocal object specifying the mode.

Reply:
So does this hard-code the collection in?

Reply:
No, this is just saying, "for the collection called 'material-theme', set the mode to the value in the parameter".

Reply:
But what if you wanted to change the collection at runtime?

Reply:
Sorry I should have been more clear. In this implementation, the way to set the mode is to use this annotation, and the way to set the theme is to use `DesignVariableCollectionProvider`. If we were to decide to go with the annotation method, I would add another annotation like `@DesignVariableCollection` to set the collection. If we were to decide to go with the composition local method, I would add another one like `DesignVariableModeProvider` to set the mode.